src/components/Signup.tsx

```diff
@@ -0,0 +1,161 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import Typography from '@mui/material/Typography';
+import Container from '@mui/material/Container';
+import Link from '@mui/material/Link';
+import Grid from '@mui/material/Grid';
+import Box from '@mui/material/Box';
+import ToggleButton from '@mui/material/ToggleButton';
+import ToggleButtonGroup from '@mui/material/ToggleButtonGroup';
+import { createUserWithEmailAndPassword } from '../api/auth';
+
+import { createFirebaseDao } from '../api/dao';
+
+function Signup() {
+    const navigate = useNavigate();
+    const [firstName, setFirstName] = useState('');
+    const [lastName, setLastName] = useState('');
+    const [password, setPassword] = useState('');
+    const [email, setEmail] = useState('');
+    const [address, setAddress] = useState('');
+    const [userType, setUserType] = useState('senior');
+
+    const validateInput = () => {
+        if (password.trim() === '' || password.length < 8) return false;
+        return true;
+    };
+
+    const handleToggle = (event: React.MouseEvent<HTMLElement>, newUserType: string) => {
+        setUserType(newUserType);
+    };
+
+    const handleSubmit = (event: { preventDefault: () => void }) => {
+        event.preventDefault();
+        if (validateInput()) {
+            createUserWithEmailAndPassword(email, password)
+                .then((registeredUser) => {
+                    if (userType === 'nurse') {
+                        const user = createFirebaseDao('nurse');
+                        user.add({
+                            uid: registeredUser.user.uid,
+                            firstName,
+                            lastName,
+                            email,
+                            address,
+                            userType,
+                        });
+                    } else {
+                        const user = createFirebaseDao('senior');
+                        user.add({
+                            uid: registeredUser.user.uid,
+                            firstName,
+                            lastName,
+                            email,
+                            address,
+                            userType,
+                        });
+                    }
+                    navigate('/');
+                })
+                .catch(() => {});
```

Comment:
Ideally, we should display something if the register was not successful. But hey, this is not work, this is only a class project :laughing:

Reply:
that's a good suggestion. I will show a success message. I still need to add some feedback if the validation fails. For now, I just redirected the user to `/`. I was thinking we could display the user schedule on the home page.